Team8923_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team8923_CENTERSTAGE/Toggle.java

```diff
@@ -0,0 +1,30 @@
+package org.firstinspires.ftc.team8923_CENTERSTAGE;
+
+public class Toggle {
+    public boolean toggleState = false;
+    public boolean prevState = false;
+    public boolean press = false;
+
+    public Toggle() {
+
+    }
+
+    public boolean toggle(boolean button) {
```

Comment:
I decided to sneak into your pull request and take a look lol
If you'd like, it's possible to make the toggle method a lot shorter and more straightforward with only 2 class variables:
```java
public boolean toggleState = false;
private boolean buttonIsHeld = false;
/**
 * flips the state of the toggle when a given button is pressed down
 * @param buttonIsPressed raw button input; pressed down event is processed internally
 * @return returns the current or resulting state of the toggle
 */
public boolean update(boolean buttonIsPressed) {
    // not held means it's a fresh press (down event)
    if (buttonIsPressed && !this.buttonIsHeld) {
        this.toggleState = !this.toggleState;
    }
    // held stays true until the button returns false
    this.buttonIsHeld = buttonIsPressed;
    return this.toggleState;
}
```
I named the method "update" instead of "toggle" because it's not always toggling the toggle but instead updating it with the current input.
Furthermore, I haven't really looked around yet for the purpose of your Toggle object but I'd recommend your constructor method take a `defaultState` parameter so that it can start as false or true depending on if you need that case.

Reply:
Oh wow hi! We actually are not even using the Toggle class anymore because we changed our controls, but your version makes sense too, thanks!